helix-term/src/commands.rs

```diff
@@ -4234,3 +4236,30 @@ fn suspend(_cx: &mut Context) {
     #[cfg(not(windows))]
     signal_hook::low_level::raise(signal_hook::consts::signal::SIGTSTP).unwrap();
 }
+
+fn add_newline_above(cx: &mut Context) {
+    add_newline_impl(cx, true);
+}
+
+fn add_newline_below(cx: &mut Context) {
+    add_newline_impl(cx, false)
+}
+
+fn add_newline_impl(cx: &mut Context, above: bool) {
+    let count = cx.count();
+    let (view, doc) = current!(cx.editor);
+    let selection = doc.selection(view.id);
+    let text = doc.text();
+    let slice = text.slice(..);
+
+    let changes = selection.into_iter().map(|range| {
+        let (start, end) = range.line_range(slice);
+        let line = if above { start } else { end + 1 };
```

Comment:
I think you'll need to make sure this doesn't panic if you `+ 1` at the last line of the document. Though that might still work since `len() + 1` is guaranteed to return the end of the document. \cc @cessen

Reply:
I'm pretty sure this won't panic.  `line_to_char(len_lines() + 1)` should always return the end-of-document position, so this should be fine.

We should make sure to test it to be sure, of course.

Reply:
It doesn't panic when I tested it in helix. Do you want me to add a unit test?